src/codegen/codegen_ispc_visitor.cpp

```diff
@@ -124,54 +124,46 @@ void CodegenIspcVisitor::visit_local_list_statement(const ast::LocalListStatemen
 /****************************************************************************************/
 
 /**
- * \todo : In ISPC we have to explicitly append `d` to a floating point number
- *         otherwise it is treated as float. A value stored in the AST can be in
- *         scientific notation and hence we can't just append `d` to the string.
- *         Hence, we have to print number with .16f and then append `d`. But note
- *         that this will result into discrepancy between C++ backend and ISPC
- *         backend when floating point number is not exactly represented with .16f.
+ * In ISPC we have to explicitly append `d` to a floating point number
+ * otherwise it is treated as float. A value stored in the AST can be in
+ * scientific notation and hence we can't just append `d` to the string.
+ * Hence, we have to transform the value into the ISPC compliant format by
+ * replacing `e` and `E` with `d` to keep the same representation of the
+ * number as in the cpp backend.
  */
 std::string CodegenIspcVisitor::double_to_string(const std::string& s_value) {
-    double value = std::stod(s_value);
-    if (std::ceil(value) == value) {
-        return "{:.1f}d"_format(value);
-    }
-    if ((value <= 1.0) && (value >= -1.0)) {
-        return "{:.16f}d"_format(value);
+    std::string return_string = s_value;
+    if (s_value.find('E') != std::string::npos) {
+        std::replace(return_string.begin(), return_string.end(), 'E', 'd');
+    } else if (s_value.find('e') != std::string::npos) {
+        std::replace(return_string.begin(), return_string.end(), 'e', 'd');
```

Comment:
You can match any char with:

```suggestion
    if (s_value.find_first_of("eE") != std::string::npos) {
        std::replace(return_string.begin(), return_string.end(), 'E', 'd');
        std::replace(return_string.begin(), return_string.end(), 'e', 'd');
```